geoportal/geoportailv3_geoportal/static-ngeo/js/mvtstyling/MvtStylingService.js

```diff
@@ -8,357 +9,391 @@ const url_get = '/get_userconfig';
-                      // for OL-Cesium to refresh the background layer counterpart
-                      // and thus request tiles with custom style
-                      this.backgroundLayerMgr_.set(map, bgLayer);
-                    });
-                } catch (error) {
-                    console.log(error);
-                    return;
-                }
-            }, 50);
+    publishIfSerial(map) {
+        const serial = new URLSearchParams(window.location.search).get('serial');
+        const isValidUUIDv4Regex = /^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/gi;
+        if (serial) {
+            // check if simple/medium styling
+            if (serial.match(isValidUUIDv4Regex) === null) {
+                const bgLayer = this.backgroundLayerMgr_.get(map);
+                const interval = setInterval(() => {
+                    try {
+                        clearInterval(interval);
+                        this.publishStyle(bgLayer).then((result) => {
+                            // for OL-Cesium to refresh the background layer counterpart
+                            // and thus request tiles with custom style
+                            this.backgroundLayerMgr_.set(map, bgLayer);
+
+                            // For deprovisionning, keep the id stored
+                            let lsData = JSON.parse(window.localStorage.getItem(bgLayer.get('label')));
+                            lsData.serial = result;
+                            window.localStorage.setItem(bgLayer.get('label'), JSON.stringify(lsData));
+                        });
+                    } catch (error) {
+                        console.log(error);
+                        return;
+                    }
+                }, 50);
+            }
         }
     }
-}
 
-unpublishIfSerial(map) {
-    const serial = new URLSearchParams(window.location.search).get('serial');
-    const isValidUUIDv4Regex = /^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/gi;
-    if (serial) {
-        // check if simple/medium styling
-        if (serial.match(isValidUUIDv4Regex) === null) {
-            const bgLayer = this.backgroundLayerMgr_.get(map);
-            if (!bgLayer) { // 3D enable watch is run before bgLayer is initialized...
-                return;
+    unpublishIfSerial(map) {
+        const serial = new URLSearchParams(window.location.search).get('serial');
+        const isValidUUIDv4Regex = /^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/gi;
+        if (serial) {
+            // check if simple/medium styling
+            if (serial.match(isValidUUIDv4Regex) === null) {
+                const bgLayer = this.backgroundLayerMgr_.get(map);
+                if (!bgLayer) { // 3D enable watch is run before bgLayer is initialized...
+                    return;
+                }
+                return this.unpublishStyle(bgLayer);
```

Comment:
I think should 3 nested `if` could be simplified for readability
(eg: `if (!serial && !isValid.test(serial)) return;`.